src/transforms/split.rs

```diff
@@ -82,6 +88,17 @@ impl Split {
             })
             .collect();
 
+        let separator = match separator {
+            Some(s) => match Regex::new(&s) {
+                Ok(expression) => Some(expression),
+                Err(err) => {
+                    error!(message = "Invalid regex expression for separator", %err);
+                    None
+                }
+            },
+            None => None,
+        };
```

Comment:
This should return an error rather than just displaying it. That way it will prevent the config from being built.